docs/docs/guides/developer_guides/smart_contracts/profiling_transactions.md

```diff
@@ -0,0 +1,79 @@
+---
+title: Profiling Transactions
+sidebar_position: 5
+tags: [contracts, profiling]
+---
+
+# Profiling Transactions
+
+An Aztec transaction typically consists of a private and a public part. The private part is where the user executes contract logic within the PXE and generates a proof of execution, which is then sent to the sequencer.
+
+Since proof generation is an expensive operation that needs to be done on the client side, it is important to optimize the private contract logic. That is, it is desirable to keep the gate count of circuits representing the private contract logic as low as possible.
+
+A private transaction can involve multiple function calls. It starts with an account `entrypoint()` which may call several private functions to execute the application logic, which in turn might call other functions. Moreover, every private function call has to go through a round of kernel circuits. Read more about the transaction lifecycle [here](../../../aztec/concepts/transactions.md).
+
+In this guide, we will look at how to profile the private execution of a transaction, allowing you to get the gate count of each private function within the transaction, including the kernel circuits.
+
+## Prerequisites
+
+- `aztec-nargo` installed (go to [Sandbox section](../../../reference/developer_references/sandbox_reference/sandbox-reference.md) for installation instructions)
+- `aztec-wallet` installed (installed as part of the Sandbox)
+- Aztec Sandbox running with **proving enabled** (go to [Sandbox PXE Proving](../local_env/sandbox_proving.md) for instructions)
+
+## Profiling using aztec-wallet
+
+The profiling tool is integrated into the `aztec-wallet`.
+
+In this example, we will profile a "private token transfer" transaction where an `operator` transfers tokens from a `user` account using [Authwit](../../../aztec/concepts/accounts/authwit.md).
```

Comment:
I don't think authwits can be used to allow users to transfer tokens on behalf of other users. I recently updated the docs [here](https://github.com/AztecProtocol/aztec-packages/pull/9438/files#diff-a2959e1391528ed214cae2c5d390ebac6d558280081221d6c66265fdc9a673b2R17).

This might work in the example because everything is on the same PXE, but if users have different PXEs I don't think it'd work.

What do you think, should this be updated to be a simple transfer?